src/main/scala/com/qubole/spark/hiveacid/writer/hive/HiveAcidWriter.scala

```diff
@@ -221,12 +221,33 @@ private[writer] class HiveAcidFullAcidWriter(options: WriterOptions,
       }
     }
 
+    def safeDeletePath(fs: FileSystem, path: Path): Boolean = {
+      try {
+        fs.delete(path, false)
+      } catch {
+        case e: Exception =>
+          logWarning(s"Error deleting $path: ${e}")
+          false
+      }
+    }
+
     if (createDelta) {
+      // Delete delta bucket file if exists. It can exist in the cases of task retries.
+      val deltaPath =  AcidUtils.createFilename(path, acidOutputFormatOptions)
+      if (deltaPath != null) {
+        safeDeletePath(fs, deltaPath)
+      }
       createVersionFile(acidOutputFormatOptions)
     }
 
     if (createDeleteDelta) {
-      createVersionFile(acidOutputFormatOptions.writingDeleteDelta(true))
+      // Delete delete_delta bucket file if it exists. It can exist in the cases of task retries.
+      val deleteDeltaOptions = acidOutputFormatOptions.clone().writingDeleteDelta(true)
+      val deleteDeltaPath = AcidUtils.createFilename(path, deleteDeltaOptions)
```

Comment:
nit: rename it to deleteDeltaBucketFilePath?

Reply:
done